vllm/config.py

```diff
@@ -1252,10 +1261,16 @@ def _get_and_verify_dtype(
         dtype = dtype.lower()
         if dtype == "auto":
             if config_dtype == torch.float32:
-                # Following the common practice, we use float16 for float32
-                # models.
-                logger.info("Casting torch.float32 to torch.float16.")
-                torch_dtype = torch.float16
+                if config.model_type == "gemma2":
+                    logger.info(
+                        "For Gemma 2, we downcast float32 to bfloat16 instead "
+                        "of float16 by default. Please specify `dtype` if you "
+                        "want to use float16.")
+                    torch_dtype = torch.bfloat16
+                else:
+                    # Following the common practice, we use float16 for float32
+                    # models.
+                    torch_dtype = torch.float16
```

Comment:
We should keep the original log about float16 casting
```suggestion
                    logger.info("Casting torch.float32 to torch.float16.")
                    torch_dtype = torch.float16
```

Reply:
It's actually duplicated with the line below:

https://github.com/vllm-project/vllm/blob/98cf2ed678580326ffc39c987304c61cb0ce4981/vllm/config.py#L1278